config-windows.md

```diff
@@ -0,0 +1,99 @@
+# Windows-specific Container Configuration
+
+This document describes the schema for the [Windows-specific section](config.md#platform-specific-configuration) of the [container configuration](config.md).
+The Windows container specification uses APIs provided by the Windows Host Compute Service (HCS) to fulfill the spec.
+
+You can configure a container's resource limits via the `resources` field of the Windows configuration.
+Do not specify `resources` unless required.
+
+
+#### Memory
+
+`memory` is an OPTIONAL configuration for the container's memory usage.
+
+The following parameters can be specified:
+
+* **`limit`** *(uint64, optional)* - sets limit of memory usage in bytes
+
+* **`reservation`** *(uint64, optional)* - sets soft limit of memory usage in bytes
```

Comment:
I didn't realize we already had that language, and think it needs to be clarified for the Linux case too.  Grepping through runC shows that `reservation` is backed by `memory.soft_limit_in_bytes`.  [Kernel docs for that][1] have the following to say about soft limits:

> .7. Soft limits
>
> Soft limits allow for greater sharing of memory. The idea behind soft limits
> is to allow control groups to use as much of the memory as needed, provided
>
> a. There is no memory contention
> b. They do not exceed their hard limit
>
> When the system detects memory contention or low memory, control groups
> are pushed back to their soft limits. If the soft limit of each control
> group is very high, they are pushed back as much as possible to make
> sure that one control group does not starve the others of memory.
>
> Please note that soft limits is a best-effort feature; it comes with
> no guarantees, but it does its best to make sure that when memory is
> heavily contended for, memory is allocated based on the soft limit
> hints/setup. Currently soft limit based reclaim is set up such that
> it gets invoked from `balance_pgdat` (`kswapd`).

So in your case I think `reservation` makes a lot of sense for the property name, but you should reroll the description to say something like “guaranteed minimum amount of memory”.

[1]: https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt